package cms_8120;
option java_package = "com.hikvision.finance.fms.cms.hpp.verify";
option java_outer_classname = "ClientVerifyProto";
option java_generic_services = true;

// 客户端登录请求
message ReqClientLogin
{
    optional int32 cmd                  = 1;    // CMD_DT_CLIENT_LOGIN
    optional int32 moudle_type          = 2;    // 登录模块类型
    optional string user_name           = 3;    // 用户账户
    optional string user_pwd            = 4;    // 用户密码
    optional string ip                  = 5;    // 登录的机器IP地址
    optional string mac                 = 6;    // MAC地址
    optional string module_version      = 7;    // 模块版本号
    optional string dc_ip               = 8;    // 数据中心服务IP
}

// 修改密码请求
message ReqChangePassword
{
    optional int32 cmd                  = 1;    // CMD_DT_CHANGE_PASSWORD
    optional int32 userId               = 2;    // 登录用户的id
    optional string oldPwd              = 3;    // 原密码
    optional string newPwd              = 4;    // 新密码
    
}

// 客户端登录响应
message RspClientLogin
{
    optional int32 cmd                      = 1;
    optional int32 result                   = 2;    // 验证结果,0-验证通过，其它值验证失败
    optional string result_info             = 3;    // 信息
    optional int32 authorization            = 4;    // 软件授权类型(0-试用,1-正式)
    optional string production_version      = 5;    // 平台版本
    optional string trial_end_time          = 6;    // 试用版到期时间
    optional int32 user_id                  = 7;    // 登录用户的id
    optional int32 role_id                  = 8;    // 登录用户的角色id, 因为用户拥有多角色，该字段废弃不用。
    optional int32 is_need_sync_time        = 9;    // 1-需要同步时间(从CMS获取时间，修改本地时间，达到本地时间与CMS时间同步)，0-不需要同步时间。
    repeated int32 ui_cs_right              = 10;   // 当前用户具有模块权限
    optional int32 is_omni_operator         = 11;   // 1-开放所有权限的用户, 0-否
    optional int32 user_level               = 12;   // 用户等级
    optional UserPwdCheck user_pwd_check    = 13;   // 用户密码保鲜
    optional string custom_title            = 14;   // 平台描述文字，用于定制客户端标题
    optional int32 skin_color               = 15[default=-1];   // 皮肤颜色:0-红黑；1-蓝白
	optional string update_server_ip        = 16;   //自动升级服务IP
	optional int32 update_server_port       = 17;   //自动升级服务端口
    optional int32 line                     = 18[default=0];   // 线路，0-内网（默认线路）；1-外网；[主要用于某些模块需要输入线路号的配置]
    optional int32 language                 = 19[default=0];   // 语言类型 0-中文
	optional string copyright               = 20;   // 版权描述信息
	optional int32 is_inital_pwd        	= 21;   // 登录使用的密码是否为初始密码(0表示不是初始化密码, 1表示初始化密码)
	optional string data_ex                	= 22;    //拓展数据	 DataEx XML
	optional string user_no                 = 23;    // 员工编号
	optional string strName                 = 24;    // 员工姓名
	optional int32 netId                    = 25;    // 网点区域Id
	optional string net                     = 26;    // 网点组织名称
	optional string version_url             = 27;    // 版本更新的url
	optional string version_no              = 28;    // CSClient最新版本号
	optional string isForceUpdate           = 29;    // CSClient强制升级 (1:启用    0:禁用)
	optional string logUpload_url           = 30;    // 日志上传的url
	optional int32 logReport                = 31;    // c/s客户端日志上报功能  (1:启用    0:禁用)
	optional int32 logReportMethod          = 32;    // c/s客户端日志上报功能开启后的上报方式  (1:ftp  0:http)
	optional string client_ip               = 33;    // 客户端ip
}

// 客户端密码保鲜信息
message UserPwdCheck
{
    optional int32 cmd                      = 1;    // CMD_RSP_CLIENT_PWD_CHECK
    optional int32 is_pwdcheck_enabled      = 2;    // 用户密码保鲜是否启用
    optional int32 remaining_days           = 3;    // 剩余天数
}


message ReqClientInfo
{
    optional int32 cmd                  = 1;    // CMD_MSG_CLIENT_LOGOUT(登出), CMD_WRITE_WORK_STATE(心跳)
    optional int32 userId               = 2;    // 登录用户的id
}

message RspClientInfo
{
	optional int32 cmd                      = 1;    // CMD_MSG_CLIENT_LOGOUT(登出), CMD_WRITE_WORK_STATE(心跳)
    optional int32 result                   = 2;    // 验证结果，0-验证通过，其它值验证失败
    optional string result_info             = 3;    // 信息
    optional int32 isModifyPwd              = 4;    // 1表示修改过PWD注销CS客户端, 0表示未修改不注销
}

service ClientVerifyService {
	rpc ClientLogin(ReqClientLogin) returns (RspClientLogin);  // CMD_DT_CLIENT_LOGIN
	rpc ClientLogout(ReqClientInfo) returns (RspClientInfo);  // CMD_MSG_CLIENT_LOGOUT
	rpc ChangePassword(ReqChangePassword) returns (RspClientLogin);    // CMD_DT_CHANGE_PASSWORD
	rpc WorkState(ReqClientInfo) returns (RspClientInfo);    // CMD_WRITE_WORK_STATE
}